/*
Copyright Â© 2022 Ci4Rail GmbH <engineering@ci4rail.com>
Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at
    http://www.apache.org/licenses/LICENSE-2.0
Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/
syntax = "proto3";
import "google/protobuf/timestamp.proto";

package bogie;
option go_package = "bogie/v1";

enum TriggerType {
  UNKNOWN = 0;
  MANUAL = 1;
  TRIGGER_CONDITION_MET = 2;
}

message VibrationSensorSamples {
  // sensor id
  int32 sensor_id = 1;
  // first sample timestamp
  google.protobuf.Timestamp first_sample_ts = 2;
  // sample rate in Hz
  double sample_rate = 3;
  // the meaurement samples
  // unit: g
  repeated float samples = 10;
}

message Bogie {
  message SteadyDrive {
    repeated double max = 1;
    repeated double rms = 2;
  }

  message Position {
    double lat = 1;
    double lon = 2;
    double alt = 3;
    double speed = 4;
    bool invalid = 5;
  }

  // bogie id
  int32 id = 1;

  // the trigger type
  TriggerType trigger_type = 2;
  // trigger timestamp
  google.protobuf.Timestamp trigger_ts = 3;

  // steady drive values at trigger moment
  SteadyDrive steady_drive = 4;
  // position at trigger moment
  Position position = 5;

  // the sensor samples
  repeated VibrationSensorSamples sensor_samples = 10;
}
